### PR TITLE
Deprecate span_near's collect_payloads

### PR DESCRIPTION
**Pull request by nik9000**
Its a noop in Lucene.

### CODE REVIEW

**Comment by nik9000** [comment]
@jpountz can you have a look at this? Its not clear to me _exactly_ how you'd load payloads now but this way isn't it.

**Comment by jpountz** [comment]
You would load payloads by passing Postings.PAYLOADS to SpanWeight.getSpans().

File: core/src/main/java/org/elasticsearch/index/query/SpanNearQueryParser.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
     public static final ParseField COLLECT_PAYLOADS_FIELD = new ParseField("collect_payloads");
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/query/SpanNearQueryParser.java]: can you make it deprecated with the 'withAllDeprecated' method?

**Comment by jpountz** [comment]
Just left a minor comment, otherwise LGTM

**Comment by nik9000** [comment]
> You would load payloads by passing Postings.PAYLOADS to SpanWeight.getSpans().

Got it. I'll make the change you recommend and then merge.

**Pull request merged**